### PR TITLE
Text editor

### LINKED ISSUE #12: Add MUI theming after core functionality to admin-portal
### Description of the issue

Project: Admin portal 
Size estimate: Medium
Priority: Low

Applying custom MUI theming has been postponed for prioritising core functionality and development of UI features. Theming depends on styling which uses client side rendering, currently the code base is trying to achieve most of the functionality via Server Side Rendering (without needing to add "use client").

It will be revisited at a later point in time when components will be converted to themed components.

### PR DESCRIPTION
**Pull request by HumzahJavid**
Closes #12 

## Description
This PR will add:

- A text editor component via monaco code editor, to allow updates of JSON specification files, with syntax highlighting and automated formatting.
- Position the text editor to the right of the instrument list.
- Add a submit button (for saving updates to the specification file).
- Add a Header for the text editor containing a title (of the chosen instrument) and the submit button.
- Add a 'client' mock Alert component which displays on button click (similar to toast from bootstrap).
- Add highlighting of chosen instrument to instrument list

### CODE REVIEW

**Review by Pasarus — CHANGES_REQUESTED** [review]
Just some comments to make, let me know if nothing is clear.

The button needs to either be at the top of the page or somewhere more sane than beneath the starting page.

Would it be possible to "prettify" the default json?

With regards to the overall layout, could the instrument list be made thinner? I think it's now obvious that it requires less space?

**Comment by HumzahJavid** [comment]
Thanks for the discussion @keiranjprice101 .
I will add a title above the code editor as you suggested, and move the button to a similar location

**Comment by HumzahJavid** [comment]
All unused (commented out) code/comments were removed in 3dcc530

**Comment by HumzahJavid** [comment]
I have updated the top level PR comment with all the changes in this PR.

File: src/components/TextEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    const updatedText = editorRef.current.getModel().getValue();
+
+    if (!scheduledChange) {
+      setScheduledChange(true);
+      console.log("scheduling formatting for code:");
+      console.log(updatedText);
+      setTimeout(() => {
+        editorRef.current.getAction("editor.action.formatDocument").run();
+        setScheduledChange(false);
+      }, 5000);
```
Comment (keiranjprice101) [inline src/components/TextEditor.tsx:28]: Can you add a comment at the end of this line like `// 5 seconds`

File: src/app/page.tsx [hunk 2]
```diff
@@ -1,5 +1,41 @@
...
+        }}
+      >
+        <Box sx={{ flex: 1 }}>
+          <InstrumentList
+            selected={instrument}
+            handleInstrumentChange={setInstrument}
+          />
+        </Box>
+        <Box sx={{ flex: 7 }}>
+          <EditorHeader title={instrument}></EditorHeader>
```
Comment (keiranjprice101) [inline src/app/page.tsx:29]: Can you put this into a single tag `<EditorHeader title={instrument} />`

**Review by keiranjprice101 — CHANGES_REQUESTED** [review]
Few minor things

**Review by keiranjprice101 — APPROVED** [review]

**Pull request merged**